packages/compass-indexes/src/components/create-index-fields.tsx

```diff
@@ -67,10 +67,21 @@ function CreateIndexFields({
   updateFieldName,
   updateFieldType,
 }: CreateIndexFieldsProps): React.ReactElement {
-  const serverHasColumnstoreIndexesSupport = useMemo(
-    () => hasColumnstoreIndexesSupport(serverVersion),
-    [serverVersion]
-  );
+  const [indexTypes, selectorWidth] = useMemo(() => {
+    const serverSupportsColumnStoreIndex =
+      hasColumnstoreIndexesSupport(serverVersion);
+    const indexTypes = serverSupportsColumnStoreIndex
+      ? INDEX_TYPES
+      : INDEX_TYPES.filter((type) => type !== 'columnstore');
```

Comment:
Maybe it’s just me, but it took me a while to parse this… I don’t know but maybe you’d find this easier to read?

```suggestion
    const indexTypes = 
      INDEX_TYPES.filter((type) => serverSupportsColumnStoreIndex || type !== 'columnstore');
```

Reply:
Ah, yeah, this reads better indeed, thank you!